jupyterhub/tests/test_scopes.py

```diff
@@ -262,79 +260,121 @@ async def test_by_fake_user(app):
 err_message = "No access to resources or resources not found"
 
 
-async def test_request_fake_user(app):
-    user_name = 'buster'
-    fake_user = 'annyong'
-    user = add_user(app.db, name=user_name)
-    test_role = generate_test_role(user_name, ['read:users!group=stuff'])
-    roles.create_role(app.db, test_role)
-    roles.grant_role(app.db, entity=user, rolename='test')
+@pytest.fixture
+def temp_role(app):
+    """Generate a temporary role with certain scopes.
+    Convenience function that takes care of db stuff and teardown"""
+    temp_roles = []
+    index = [1]
+
+    def temp_role_creator(scopes, role_name=None):
+        if not role_name:
+            role_name = f'test_role_{index[0]}'
+            index[0] += 1
```

Comment:
You can use `nonlocal counter` to declare a shared variable from the inherited scope

rather than relying on a list for the closure:

```suggestion
            nonlocal counter
            role_name = f'test_role_{counter}'
            counter += 1
```
```

Reply:
Much nicer, thanks!